readthedocs/projects/tasks.py

```diff
@@ -1817,37 +1818,67 @@ def finish_inactive_builds():
     """
     Finish inactive builds.
 
-    A build is consider inactive if it's not in ``FINISHED`` state and it has been
-    "running" for more time that the allowed one (``Project.container_time_limit``
-    or ``DOCKER_LIMITS['time']`` plus a 20% of it).
+    A build is consider inactive if all the following are true:
 
-    These inactive builds will be marked as ``success`` and ``FINISHED`` with an
+    - it's not in ``FINISHED`` state
+    - it was created +15 minutes ago
```

Comment:
I think we can reduce this time. If the task is not queued nor being executed, it already means that it's inactive this particular build won't change its state.